service-base/src/main/java/org/eclipse/hono/service/management/credentials/EventBusCredentialsManagementAdapter.java

```diff
@@ -94,31 +95,27 @@ protected String getEventBusAddress() {
         final JsonObject payload = request.getJsonPayload();
         final SpanContext spanContext = request.getSpanContext();
 
+        final Span span = Util.newChildSpan(SPAN_NAME_UPDATE_CREDENTIAL, spanContext, tracer, tenantId, deviceId, getClass().getSimpleName());
+
+        final Future<EventBusMessage> resultFuture;
         if (tenantId == null) {
-            return Future.failedFuture(new ClientErrorException(
-                    HttpURLConnection.HTTP_BAD_REQUEST,
-                    "missing tenant ID"));
+            log.debug("missing tenant ID");
+            TracingHelper.logError(span, "missing tenant ID");
+            resultFuture = Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, "missing tenant ID"));
         } else if (payload == null) {
-            return Future.failedFuture(new ClientErrorException(
-                    HttpURLConnection.HTTP_BAD_REQUEST,
-                    "missing payload"));
-        }
-        try {
-            final Future<List<CommonCredential>> secretsFuture = credentialsFromPayload(request);
-
-            final Span span = Util.newChildSpan(SPAN_NAME_UPDATE_CREDENTIAL, spanContext, tracer, tenantId, deviceId, getClass().getSimpleName());
-            final Promise<OperationResult<Void>> result = Promise.promise();
-
-            return secretsFuture.compose(secrets -> {
-                getService().set(tenantId, deviceId, resourceVersion, secrets, span, result);
-                        return result.future().map(res -> {
-                            return res.createResponse(request, id -> null).setDeviceId(deviceId);
-                        });
-                    }
-            );
-        } catch (final IllegalStateException e) {
```

Comment:
Note that this `IllegalStateException` (which should rather be an `IllegalArgumentException`, see added TODO) is now caught in `credentialsFromPayload()`.